WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -313,7 +313,7 @@ def my_competitions
     if current_user.person
       @competitions += current_user.person.competitions
     end
-    @competitions = @competitions.uniq.sort_by(&:start_date).reverse!
+    @competitions = @competitions.uniq.reject(&:start_date) + @competitions.uniq.select(&:start_date).sort_by(&:start_date).reverse!
```

Comment:
Nice, I was going to use:

@competitions = @competitions.uniq.sort_by { |comp| comp.start_date || Date.today + 20.year }.reverse